### PR TITLE
[WIP] Test TF Flaubert + Add {XLM, Flaubert}{TokenClassification, MultipleC…

### PR DESCRIPTION
**Pull request by LysandreJik**
…hoice} models and tests

The remaining TF tests pass with TF2.3. Waiting to unpin TF before merge.

### CODE REVIEW

File: src/transformers/modeling_tf_flaubert.py [hunk 1]
```diff
@@ -136,8 +139,8 @@ def call(
         head_mask=None,
         inputs_embeds=None,
         training=False,
```
Comment (jplu) [inline src/transformers/modeling_tf_flaubert.py:141]: I would suggest to put `training` at the end.

File: src/transformers/modeling_tf_flaubert.py [hunk 2]
```diff
@@ -257,9 +267,12 @@ def call(
 
             # self attention
             if not self.pre_norm:
-                attn_outputs = self.attentions[i]([tensor, attn_mask, None, cache, head_mask[i]], training=training)
+                attn_outputs = self.attentions[i](
+                    [tensor, attn_mask, None, cache, head_mask[i], output_attentions], training=training
+                )
                 attn = attn_outputs[0]
-                attentions = attentions + (attn_outputs[1],)
+                if output_attentions:
```
Comment (jplu) [inline src/transformers/modeling_tf_flaubert.py:274]: Replace by:
```
if cast_bool_to_primitive(output_attentions, self.output_attentions) is True:
```
Same at lines 285 and 321.

Replace as well:
```
if output_hidden_states:
```
By
```
if cast_bool_to_primitive(output_hidden_states, self.output_hidden_states) is True:
```
At lines 265, 308, and 319

**Review by jplu — COMMENTED** [review]
Can you try to change line 261:
```
dropout_probability = random.uniform(0, 1)
```
Into
```
dropout_probability = tf.random.uniform([], 0, 1)
```
To see if it changes something?

File: tests/test_modeling_common.py [hunk 3]
```diff
@@ -140,6 +140,7 @@ def test_attention_outputs(self):
             encoder_seq_length = encoder_seq_length * self.model_tester.num_hashes
 
         for model_class in self.all_model_classes:
+            print(model_class)
```
Comment (sgugger) [inline tests/test_modeling_common.py:143]: Left from debugging?

File: tests/test_modeling_flaubert.py [hunk 4]
```diff
@@ -319,6 +328,38 @@ def create_and_check_flaubert_token_classif(
         self.parent.assertListEqual(list(result["logits"].size()), [self.batch_size, self.seq_length, self.num_labels])
         self.check_loss_output(result)
 
+    def create_and_check_bert_for_multiple_choice(
```
Comment (sgugger) [inline tests/test_modeling_flaubert.py:331]:
```suggestion
    def create_and_check_flaubert_for_multiple_choice(
```

File: tests/test_modeling_tf_flaubert.py [hunk 5]
```diff
@@ -18,11 +18,341 @@
...
+            "lengths": input_lengths,
+        }
+        return config, inputs_dict
+
+
+@require_tf
+class TFFlaubertModelTest(TFModelTesterMixin, unittest.TestCase):
+
+    all_model_classes = (
+        # TODO The multiple choice model is missing and should be added.
```
Comment (sgugger) [inline tests/test_modeling_tf_flaubert.py:304]: The TODO seems outdated.

**Review by sgugger — APPROVED** [review]
A few nits to clean up ut looks great otherwise!

File: tests/test_modeling_common.py (hunk 3 above)
Comment (LysandreJik) [reply tests/test_modeling_common.py:143]: oops

**Pull request merged**